target/ppc/translate/vector-impl.c.inc

```diff
@@ -25,27 +25,27 @@
         }                                               \
     } while (0)
 
-static bool trans_VCFUGED(DisasContext *ctx, arg_VX *a)
+typedef void (*VXDoublewordHelper)(TCGv_i64 tgt, TCGv_i64 src, TCGv_i64 mask);
+
+static bool do_call_vx_doubleword_helper(DisasContext *ctx, arg_VX *a,
+                                         VXDoublewordHelper gen_helper)
 {
     TCGv_i64 tgt, src, mask;
 
-    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
-    REQUIRE_ALTIVEC(ctx);
-
     tgt = tcg_temp_new_i64();
     src = tcg_temp_new_i64();
```

Comment:
Since this is now a generic utility function for dealing with Vector Facility instructions and VX-form, maybe rename 'src' and 'mask' to generic names (arg1+arg2, arg_a+arg_b, or something similar)

Reply:
I would also put the change to VCFUGED and the introduction of do_call_vx_doubleword_helper in a separate commit, unrelated to vpdepd/vpextd.

Reply:
At first glance it seemed that all insn that could use this method had source and mask registers, but a more generic name might be future proof, Ill change that.

I did this in a single commit to avoid introducing an "useless" method, but I can I split it too.